### ISSUE TITLE
--save flag while installing new packages?

### ISSUE THREAD

**Issue by cdax** [pypa/pipfile#19]
In the `npm` world, you can install new packages like so:

`npm install --save <package>`

and the `--save` flag will write the package details to the `package.json` (`pipfile` equivalent). I feel this will be a neat inclusion for `pip`/`pipfile`

**Comment by defnull** [comment]
Only possible if `Pipfile` (not only `Pipfile.lock`) uses a parseable standard format and there exists a library that supports lossless editing (i.e. load->edit->save while preserving comments and whitespace). See #8

**Comment by iamale** [comment]
I think, in fact, that this should be the only option (like in [yarn](https://yarnpkg.com/)) — Pipfile should always be up-to-date with installed packages. This is only an IMHO, though.

**Comment by kennethreitz** [comment]
Pipfiles are to be manually written, not generated.

**Comment by kennethreitz** [comment]
@iamale that would be too much of a departure from current pip behavior, in my opinion.

**Comment by kennethreitz** [comment]
i will think about it, though. the lockfile being always up-to-date is the important mechanism here.

**Comment by reillysiemens** [comment]
@iamale: I, for one, would not like to see that behavior.

For example, let's say I'm working on a project that has some dependencies. I have an existing `Pipfile` that is aligned with those dependencies. Now I go ahead and install something like IPython for exploring my environment.

I don't want IPython listed in my dependencies. I don't even want it listed in my development or testing dependencies since I'm the only one using it. It should be perfectly acceptable to have packages installed into your environment that aren't reflected in the `Pipfile`.

**Comment by pradyunsg** [comment]
Since #47, this is mostly simpler to implement now. The package can be added pinned to the installed version.

Maybe allow `--save` to take an argument - the group to add the package into (default, dev etc)?

**Comment by kennethreitz** [comment]
Pipenv does this.

**Comment by brokenthorn** [comment]
> Pipenv does this.

Let me rephrase that. Use `pipenv install <package_name>`, not `pip install <package_name>` for the equivalent of `npm install <package_name>` (`--save` is optional now with npm and implied by default).
